node-db/lesson2.md

```diff
@@ -1,22 +1,392 @@
-# Node II
+![](https://img.shields.io/badge/status-draft-darkred.svg) 
+# Node 1
+**What we will learn today?**
+- Recap
+- Templating
+    - MVC
+- Callbacks
+    - callback hell
+    - promises
+- Middleware and Error handling
+- Deploying to Heroku
 
-### It's our second week of Node.js! The plan is as follows:
+---
+ToDo: Clone the repo for express-workshop-2
 
-## 10:30 AM - Homework review
+# Recap
 
-## 11:00 AM - Scrum session with Kram
+- **Package.json**: Remember how in the last lesson we added a *package.json* file and added dependencies like *express* to it.
 
-## 1:00 PM - Node session
+    - What command did we use to create a *package.json* file
+    - How did we add dependencies to it?
 
-This is going to be a mixture of instructor-led parts to recap the fundamentals of Node.js, and practical pair-programming parts to get more experience with building Node.js apps.
+    Today, the project comes with a package file already - this lists all the packages we need to run the project (instead of adding installed packages to the Git repo, which would take up a lot of memory, we just declare a list of dependencies in `package.json`). Now, we can run `npm install`, and the dependencies will be installed.
 
-### Node.js Fundamentals Recap
+- **Server.js** - this is the entry point for our application. Let's walk through the code and remember what it does.
+    - What is **Express Static Middleware** used for?
+    - What routes are defined?
+    - What **Port** does the server run on?
 
-[Link](/node-recap.md)
+> Let's run the server and check it in the browser. This is what you should see on your browser:
+![](assets/blog-screenshot.png)
 
-### Practical exercises: The Status Code API and The Status Code Website.
+* The theme used for this website is based on Bootstrap. You can get more open source Bootsrap-based themes from [StartBootstrap](https://startbootstrap.com/template-categories/all/)
 
-Part 1. [Build a Node status code API](https://github.com/Code-Your-Future/node-status-code-api)
+> **Exercise**: Notice how the link to **Contact me** doesn't return a page - Let's add an enpoint to return a simple string **this is a contact page**.
```

Comment:
In the homework they've already been asked to return a page, so they should know how to do that. Should we just ask them to return a full page rather than just a string?